.github/actions/build_image/action.yml

```diff
@@ -84,11 +125,26 @@ runs:
       id: image_version_branch
       run: |
         base_image=$(echo ${{ inputs.image_name }} | cut -d '-' -f1)
-        app_version=$(podman run --pull=never pulp/${{ inputs.image_name }}:ci-amd64 bash -c "pip3 show pulpcore | sed -n -e 's/Version: //p'")
+        podman run --pull=never pulp/${{ inputs.image_name }}:ci-amd64 bash -c "pip3 freeze | grep pulp" >> versions.freeze
+        app_version=$(grep pulpcore versions.freeze | sed -n -e 's/pulpcore==//p')
         app_branch=$(echo ${app_version} | grep -oP '\d+\.\d+')
 
         echo "APP_VERSION: ${app_version}"
         echo "APP_BRANCH: ${app_branch}"
         echo "app_version=${app_version}" >> "$GITHUB_OUTPUT"
         echo "app_branch=${app_branch}" >> "$GITHUB_OUTPUT"
       shell: bash
+
+    - name: Clear cache for next upload
+      if: env.BUILD == 'true' && steps.cache.outputs.cache-hit == 'true' && github.event_name != 'pull_request'
+      run: |
+        echo "Deleting existing cache for ${{ env.VERSIONKEY }}"
+        gh cache delete "base-images=${{ env.VERSIONKEY }}" -R ${{ github.repository }}
+      shell: bash
+
+    - name: Cache versions
+      if: env.BUILD == 'true' && github.event_name != 'pull_request'
+      uses: actions/cache/save@v4
+      with:
+        key: ${{ env.VERSIONKEY }}
```

Comment:
Is this the same cache that we key with `base_images={{ env.VERSIONKEY }}`?

Reply:
It's the same cache container, but not the same entry. `base_images={{ env.VERSIONKEY }}` was a typo left over from copy-pasting. I fixed it, now the key should only be `{{ env.VERSIONKEY }}` for the `versions.freeze` file.